### PR TITLE
Add Ruff check GitHub Actions push and pull request merge block

### LINKED ISSUE #35: Ruff linting and formatting merge check
That would be great too. I think we can set up `pre-commit` to run on commit and add a merge check too just in case. Perhaps we should open a separate issue for further discussion?

_Originally posted by @n0Oo0Oo0b in https://github.com/ib-ai/ib.py/issues/33#issuecomment-2123856467_

### PR DESCRIPTION
**Pull request by EdwardTheLegend**
Added the `ruff.yml` for the GitHub Actions workflow and also had to force fixes to solve linter violations such as removing unused imports.

Closes #35

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -6,46 +6,52 @@
...
 
 intents = discord.Intents.all()
 INITIAL_COGS = (
-    'dev',
-    'embeds',
-    'filter',
+    "dev",
+    "embeds",
+    "filter",
     # 'help',
```
Comment (ilikecubesnstuff) [inline bot.py]: I assume Ruff missed this because it's commented out, but change these quotation marks as well.

File: cogs/monitor.py [hunk 2]
```diff
@@ -183,34 +213,42 @@ async def message_delete(self, ctx: commands.Context, pattern_id: int):
...
         await monitored_message.save()
         get_all_monitor_messages.cache_clear()
 
-        logger.debug(f"{'Disabled' if monitored_message.disabled else 'Enabled'} pattern with ID {monitored_message.monitor_message_id} in monitor.")
-        await ctx.send(f"The pattern (`{monitored_message.message}`) [ID: {monitored_message.monitor_message_id}] has been successfully {'disabled' if monitored_message.disabled else 'enabled'}.")
+        logger.debug(
+            f"{'Disabled' if monitored_message.disabled else 'Enabled'} pattern with ID {monitored_message.monitor_message_id} in monitor."
+        )
+        await ctx.send(
+            f"The pattern (`{monitored_message.message}`) [ID: {monitored_message.monitor_message_id}] has been successfully {'disabled' if monitored_message.disabled else 'enabled'}."
```
Comment (NathanealV) [inline cogs/monitor.py]: Line 241 + line 305 are over the 95 character line length (may involve other lines as well). Should probably manually adjust these.

**Review by NathanealV — COMMENTED** [review]
Thank you for the PR! Changes look good - might need to go over some files again with f-strings as they might be over the line length (mostly with `cogs/monitor.py`.

**Comment by EdwardTheLegend** [comment]
I ran the formatting with the new line length on the whole repo. I think the exceptions you mentioned might be where the ruff rules based on pep standards don't recommend folding on. Some things like long strings aren't usually broken over multiple lines even if they go a bit over the limit.

**Review by ilikecubesnstuff — CHANGES_REQUESTED** [review]
Apart from my comment and Nath's, this looks fine now. I'll approve after the changes are pushed. Thanks!

**Review by ilikecubesnstuff — APPROVED** [review]

**Review by NathanealV — APPROVED** [review]
LGTM!

**Pull request merged**